unity/ARUI/Assets/Scripts/ResearchModeCapture.cs

```diff
@@ -58,69 +64,159 @@ enum DepthSensorMode
         return ref this._logger;
     }
 
-    private void Awake()
+    /// <summary>
+    /// Initializes the depth camera sensor.
+    /// </summary>
+    void Awake()
     {
+        Logger log = logger();
+
 #if ENABLE_WINMD_SUPPORT
-        unityWorldOrigin = Windows.Perception.Spatial.SpatialLocator.GetDefault().CreateStationaryFrameOfReferenceAtCurrentLocation().CoordinateSystem;
+        this.sensorDevice = new ResearchModeSensorDevice();
+        this.requestCameraAccessTask = this.sensorDevice.RequestCameraAccessAsync().AsTask();
+        this.cameraSensor = (ResearchModeCameraSensor)this.sensorDevice.GetSensor(ResearchModeSensorType.DepthAhat);
+
+        Guid rigNodeGuid = this.sensorDevice.GetRigNodeId();
+        this.rigNodeLocator = SpatialGraphInteropPreview.CreateLocatorForNode(rigNodeGuid);
 #endif
     }
 
-    // Start is called before the first frame update
+    /// <summary>
+    /// Checks that the depth camera was initialized properly,
+    /// creates the ROS publisher for the depth camera,
+    /// and starts the camera capturing thread.
+    /// </summary>
     void Start()
     {
-        // TODO: Add this back in and use ROS-TCP plugin instead when we need research mode sensors
-
-        /*
         Logger log = logger();
 
-        try
-        {
-            TcpServerIPAddr = PTGUtilities.getIPv4AddressString();
-        }
-        catch (InvalidIPConfiguration e)
-        {
-            log.LogInfo(e.ToString());
-            return;
-        }
+#if ENABLE_WINMD_SUPPORT
+        var consent = this.requestCameraAccessTask.Result;
```

Comment:
Is it possible to just access "Result" without assigning it to an unused variable? (which I assume is a build warning)

Reply:
Ooops, I actually meant to have a check to make sure that the camera was accessed OK. It's there now.